nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import errno
+import argparse
+
+def parse_args(args=None):
+    Description = 'Reformat {{ cookiecutter.name }} samplesheet file and check its contents.'
+    Epilog = """Example usage: python check_samplesheet.py <FILE_IN> <FILE_OUT>"""
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument('FILE_IN', help="Input samplesheet file.")
+    parser.add_argument('FILE_OUT', help="Output file.")
```

Comment:
I don't really understand why you print out a new samplesheet? Why can you not just validate the one that exists and exit with an error if there's a problem?

Reply:
So this is there to validate the input samplesheet as well as to reformat it to be used in the pipeline. For example, the output samplesheet will have a `single_end` parameter too which would have been auto-detected from the input samplesheet. Also, I tend to rename the `sample` field for instances where you may want to merge multiple fastq files from the same sample. Its much easier to do everything this way and then just read in the output samplesheet in the way that NF needs it for the pipeline.